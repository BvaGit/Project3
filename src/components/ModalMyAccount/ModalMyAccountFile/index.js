import ModalMyAccountFile from './ModalMyAccountFile';
import { connect } from 'react-redux';
import { avatarPUT, myAccountGET } from '../../../store/user/asyncActions';
import { getUserAvatar } from '../../../store/user/selectors';

const mapStateToProps = (state) => ({
    avatar: getUserAvatar(state),
});

const mapDispatchToProps = (dispatch) => ({
    upAvatar: (data) => dispatch(avatarPUT(data)),
    getMyAccount: () => dispatch(myAccountGET())
})


export default connect(mapStateToProps, mapDispatchToProps)(ModalMyAccountFile);