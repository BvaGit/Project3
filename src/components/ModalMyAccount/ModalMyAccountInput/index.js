import { connect } from "react-redux";
import ModalMyAccountInput from "./ModalMyAccountInput";

import { changeMyAccountField } from '../../../store/users/actions';


const mapStateToProps = (state) => ({})

const mapDispatchToProps = (dispatch) => ({
    changeMyAccountField: (data) => dispatch(changeMyAccountField(data)),
})

export default connect(mapStateToProps,mapDispatchToProps)(ModalMyAccountInput);