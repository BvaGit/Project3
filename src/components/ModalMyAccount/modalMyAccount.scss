@import '../../utils/variables.scss';

.my-account{
    @include flex(column,_,_);
    &__avatar{
        text-align: center;
    }
    &__info{
        @include flex(_,space-between,_);
        margin-bottom: $margin_md;
    }
    &__infochild{
        width: 45%;
    }
    &__input{
        width: 100%;
        border: none;
        padding: $padding_md;
        background:rgb(226, 245, 253);
        outline-color: $link-color;
    }
    &__input::placeholder{
        color: $title-color;
        font-style: italic;
        font-size: $margin_md;
    }
}
