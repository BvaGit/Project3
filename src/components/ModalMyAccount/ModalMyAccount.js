import React from 'react';
import { FormattedMessage } from 'react-intl';

import Input from '../Input';
import message from '../constants/localization';

import './modalMyAccount.scss';
import '../RoomsHeader/roomsHeader.scss';

function ModalMyAccount({languageValue: langValue, handlecloseModal}){
    return (
        <div className="my-account">
                <div className="my-account__avatar">
                    <img className="rooms-header__avatar" src="../../../public/assets/images/1-36-1024x1024.jpg"/>
                </div>
                <div className="my-account__info">
                    <div className="my-account__info1">
                        <label className="inputs__label">
                        <FormattedMessage 
                            id="firstName" 
                            defaultMessage="First name" 
                            value={{ langValue }}>
                            First name
                        </FormattedMessage>
                        <FormattedMessage 
                            id="firstNamePlaceholder" 
                            defaultMessage="Enter your first name..." 
                            value={{ langValue }}>
                            {placeholder=>
                                <Input type="text" placeholder={placeholder}/> 
                            }
                        </FormattedMessage>
                        </label>
                    </div>
                    <div className="my-account__info1">
                        <label className="inputs__label">
                        <FormattedMessage 
                            id="lastName" 
                            defaultMessage="Last name" 
                            value={{ langValue }}>
                            Last name
                        </FormattedMessage>
                        <FormattedMessage 
                            id="lastNamePlaceholder" 
                            defaultMessage="Enter your last name..." 
                            value={{ langValue }}>
                            {placeholder=>
                            <Input type="text" placeholder={placeholder}/>
                            }
                        </FormattedMessage>
                        </label>
                    </div>
                </div>
                <div className="my-account__info">
                    <div className="my-account__info1">
                        <label className="inputs__label">
                        <FormattedMessage 
                            id="age" 
                            defaultMessage="Age" 
                            value={{ langValue }}>
                            Age
                        </FormattedMessage>
                        <FormattedMessage 
                            id="agePlaceholder" 
                            defaultMessage="Enter your age..." 
                            value={{ langValue }}>
                            {placeholder=>
                            <Input type="number" placeholder={placeholder}/>
                            }
                        </FormattedMessage>
                        </label>
                    </div>
                    <div className="my-account__info1">
                        <label className="inputs__label">
                        <FormattedMessage 
                            id="city" 
                            defaultMessage="City" 
                            value={{ langValue }}>
                            City
                        </FormattedMessage>
                        <FormattedMessage 
                            id="cityPlaceholder" 
                            defaultMessage="Enter your city..." 
                            value={{ langValue }}>
                            {placeholder=>
                            <Input type="text" placeholder={placeholder}/>
                            }
                        </FormattedMessage>
                        </label>
                    </div>
                </div>
                <div className="my-account__info">
                    <div className="my-account__info1">
                        <label className="inputs__label">
                        <FormattedMessage 
                            id="companyName" 
                            defaultMessage="Company name" 
                            value={{ langValue }}>
                            Company name
                        </FormattedMessage>
                        <FormattedMessage 
                            id="companyNamePlaceholder" 
                            defaultMessage="Enter your company name..." 
                            value={{ langValue }}>
                            {placeholder=>
                            <Input type="text" placeholder={placeholder}/>
                            }
                        </FormattedMessage>
                        </label>
                    </div>
                    <div className="my-account__info1">
                        <label className="inputs__label">
                        <FormattedMessage 
                            id="hobby" 
                            defaultMessage="Hobby" 
                            value={{ langValue }}>
                            Hobby
                        </FormattedMessage>
                        <FormattedMessage 
                            id="hobbyPlaceholder" 
                            defaultMessage="Enter your hobby..." 
                            value={{ langValue }}>
                            {placeholder=>
                            <Input type="text" placeholder={placeholder}/>
                            }
                        </FormattedMessage>
                        </label>
                    </div>
                </div>
                <div className="button-box">
                    <button className="button">
                    <FormattedMessage 
                        id="saveBTN" 
                        defaultMessage="Save" 
                        value={{ langValue }}>
                        Save
                    </FormattedMessage>
                    </button>
                    <button className="button" onClick={handlecloseModal}>
                    <FormattedMessage 
                        id="cancelBTN" 
                        defaultMessage="Cancel" 
                        value={{ langValue }}>
                        Cancel
                    </FormattedMessage>
                    </button>
                </div>
        </div>
    )
}

export default ModalMyAccount;
