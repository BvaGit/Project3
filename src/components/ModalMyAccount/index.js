import ModalMyAccount from "./ModalMyAccount";
import { connect } from 'react-redux'

import { myAccountPOST, myAccountGET } from '../../store/user/asyncActions';
import { getUserAvatar } from '../../store/user/selectors';

const mapStateToProps = (state) => ({
    avatar: getUserAvatar(state)
})

const mapDispatchToProps = (dispatch) => ({
    postMyAccont: () => dispatch(myAccountPOST()),
    getMyAccount: () => dispatch(myAccountGET())
    
})

export default connect(mapStateToProps, mapDispatchToProps)(ModalMyAccount);