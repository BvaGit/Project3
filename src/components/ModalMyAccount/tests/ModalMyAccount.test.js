import React from 'react';
import { shallow, mount } from 'enzyme';
import ModalMyAccount from '../ModalMyAccount';
import { IntlProvider } from "react-intl";
import { Provider } from 'react-redux';


describe('ModalMyAccount', () => {
    it("Should match snapshot", () => {
        const component = shallowBiggerSmart(<ModalMyAccount />)
        expect(component.html()).toMatchSnapshot()
    })
})

export const shallowBiggerSmart = (component, store) => {
    const core = store
        ? <Provider store={store}>{component}</Provider>
        : component;
    return shallow(<IntlProvider>core</IntlProvider>);
};
