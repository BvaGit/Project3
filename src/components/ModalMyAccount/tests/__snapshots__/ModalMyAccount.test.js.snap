// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalMyAccount Should match snapshot 1`] = `"<div class=\\"my-account\\"><label class=\\"load-img-file\\"><img src=\\"/assets/images/1-36-1024x1024.jpg\\"/><input type=\\"file\\" name=\\"filedata\\" hidden=\\"\\"/></label><div class=\\"my-account__info\\"><div class=\\"my-account__info1\\"><label class=\\"inputs__label \\">firstName<input type=\\"text\\" name=\\"firstname\\" title=\\"firstNamePlaceholder\\" placeholder=\\"firstNamePlaceholder\\" class=\\"inputs__input undefined\\"/></label></div><div class=\\"my-account__info1\\"><label class=\\"inputs__label \\">lastName<input type=\\"text\\" name=\\"lastname\\" title=\\"lastNamePlaceholder\\" placeholder=\\"lastNamePlaceholder\\" class=\\"inputs__input undefined\\"/></label></div></div><div class=\\"my-account__info\\"><div class=\\"my-account__info1\\"><label class=\\"inputs__label \\">age<input type=\\"text\\" name=\\"age\\" title=\\"agePlaceholder\\" placeholder=\\"agePlaceholder\\" class=\\"inputs__input undefined\\"/></label></div><div class=\\"my-account__info1\\"><label class=\\"inputs__label \\">city<input type=\\"text\\" name=\\"city\\" title=\\"cityPlaceholder\\" placeholder=\\"cityPlaceholder\\" class=\\"inputs__input undefined\\"/></label></div></div><div class=\\"my-account__info\\"><div class=\\"my-account__info1\\"><label class=\\"inputs__label \\">companyName<input type=\\"text\\" name=\\"company\\" title=\\"Enter your company name...\\" placeholder=\\"Enter your company name...\\" class=\\"inputs__input undefined\\"/></label></div><div class=\\"my-account__info1\\"><label class=\\"inputs__label \\">hobby<input type=\\"text\\" name=\\"hobby\\" title=\\"hobbyPlaceholder\\" placeholder=\\"hobbyPlaceholder\\" class=\\"inputs__input undefined\\"/></label></div></div><div class=\\"button-box\\"><button class=\\"button\\">Save</button><button class=\\"button\\">Cancel</button></div></div>"`;
